pkg/packetfilter/fake/iptables.go

```diff
@@ -24,46 +24,84 @@ import (
 	"sync"
 
 	. "github.com/onsi/gomega"
-	"github.com/submariner-io/submariner/pkg/iptables"
+	fakeipset "github.com/submariner-io/submariner/pkg/ipset/fake"
+	"github.com/submariner-io/submariner/pkg/packetfilter"
+	"github.com/submariner-io/submariner/pkg/packetfilter/iptables"
 	"k8s.io/utils/set"
 )
 
-type basicType struct {
+var driverInstance *FIPTablesImpl
+
+const DriverName = "fake"
+
+type basicIptablesType struct {
 	mutex                    sync.Mutex
 	chainRules               map[string]set.Set[string]
 	tableChains              map[string]set.Set[string]
 	failOnAppendRuleMatchers []interface{}
 	failOnDeleteRuleMatchers []interface{}
 }
 
-type IPTables struct {
-	iptables.Adapter
+type FIPTablesImpl struct {
+	*iptables.IPTablesImpl
+}
+
+func SetAsDriver() *FIPTablesImpl {
+	packetfilter.AddDriver(DriverName, NewDriver)
+	packetfilter.SetDriver(DriverName)
+
+	pBasic := &basicIptablesType{
+		chainRules:  map[string]set.Set[string]{},
+		tableChains: map[string]set.Set[string]{},
+	}
+
+	a := &iptables.IPTablesImpl{
+		Basic:      pBasic,
+		IPSetIface: fakeipset.New(),
+	}
+	a.Init()
+	driverInstance = &FIPTablesImpl{a}
+
+	return driverInstance
+}
+
+func ClearDriver() {
+	driverInstance = nil
 }
 
-func New() *IPTables {
-	return &IPTables{
-		Adapter: iptables.Adapter{
-			Basic: &basicType{
-				chainRules:  map[string]set.Set[string]{},
-				tableChains: map[string]set.Set[string]{},
-			},
-		},
+func NewDriver() (packetfilter.Driver, error) {
+	if driverInstance != nil {
+		return driverInstance, nil
+	}
+
+	pBasic := &basicIptablesType{
+		chainRules:  map[string]set.Set[string]{},
+		tableChains: map[string]set.Set[string]{},
+	}
+
+	a := &iptables.IPTablesImpl{
```

Comment:
This should be a fake implementation of the packetfilter.Interface and not implement/reference/use iptables, ie it should be a sibling of iptables.